tests/test_unique_ids.py

```diff
@@ -0,0 +1,35 @@
+from bs4 import BeautifulSoup  # type: ignore
+from typer.testing import CliRunner
+from jupyter_book_to_htmlbook.xref_processing import process_ids
+
+runner = CliRunner()
+
+
+def test_unique_ids():
+    """
+    IDs should be unique; duplicated IDs should be numbered sequentially
+    and those changes should be logged. Given a list, duplicate ids should
+    be updated (with randomized number) and then returned to the list
+    """
+    existing_ids = ["foo"]
+    start = BeautifulSoup("""<h1 id="foo">Hello</h1>""", "html.parser")
+    result, additions = process_ids(start, existing_ids)
+    assert len(result.find_all(id="foo")) == 0
+    assert len(additions) == 1
+
+
+def test_xrefs_are_updated_when_ids_change():
+    """
+    If an id is changed, we should search for any internal references to it
+    and update that cross ref with the new ID.
+
+    The assumption is that any reference to a duplicate ID supposes that the
+    ID in the same file is meant.
+    """
+    existing_ids = ["foo"]
+    start = BeautifulSoup("""<h1 id="foo">Hello</h1>
+            <p><a href="#foo">link to heading</a></p>
+            <p><a href="#foo">another link to heading</a></p>
+            """, "html.parser")
+    result, additions = process_ids(start, existing_ids)
+    assert len(result.find_all(href="#foo")) == 0
```

Comment:
Thought: Could we use regex to add another `assert` to these tests to check that the new `href` values are `foo` + a random int? It maybe overkill, but that could protect against a situation where the original `id`/`href` was being removed but the substitution failed in which case these tests as is would still pass.

Reply:
This is a good point, and can be applied to the other test as well. Shouldn't have been lazy...